tests/test_visitors/test_ast/test_conditionals/test_named_constants.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+
+from wemake_python_styleguide.violations.consistency import (
+    NamedConstantConditionalViolation,
+)
+from wemake_python_styleguide.visitors.ast.conditionals import (
+    NamedConstantConditionalVisitor,
+)
+
+if_statement = """
```

Comment:
We also need to test:
1. `while` statements (`while True` is popular usecase that can be used)
2. ternary expressions
3. `if` inside the comprehensions: `[x for x in [1, 2, 3] if True]` should be forbidden

Reply:
Are you sure that there shouldn't be support for case no 1. ?

Reply:
@SheldonNunes I am really not sure. What do you think?

Reply:
I can see some use cases for `while True` especially around polling. I was thinking wrapping the boolean value in a variable may be what we want to enforce. However I can see a situation where a variable is set to `True` and we have a break clause inside to prevent running a potential large block of code. The variable would provide no real benefit as it wouldn't be modified and could potentially confuse the reader.

Reply:
Ok, let's leave `while` out of scope for now.